src/__tests__/CardDetail.test.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+  MemoryRouter,
+  RouterProvider,
+  createMemoryRouter,
+} from 'react-router-dom';
+import { act, fireEvent, render, screen } from '@testing-library/react';
+import { routes } from '../router/router';
+import { fakeData } from './fakeData/fakeData';
+import CardDetail from '../components/card-detail/CardDetail';
+
+describe('Detailed card tests', () => {
+  beforeEach(() => {
+    vi.mock('../api/api', () => {
+      return {
+        getSpell: vi.fn(async () => {
+          return fakeData;
+        }),
+        findSpells: vi.fn(async () => {
+          return fakeData;
+        }),
+      };
+    });
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+    vi.resetAllMocks();
+  });
+
+  test('Make sure the detailed card component correctly displays the detailed card data', async () => {
```

Comment:
Better without 'Make sure'
Also I think the test is big. It's better to create separate tests to make sure card data components are rendered